tests/testthat/test-eval_code.R

```diff
@@ -0,0 +1,97 @@
+testthat::test_that("eval_code evaluates the code in the quosures environment", {
+  q1 <- new_quosure("iris1 <- iris", env = list2env(list(iris1 = iris)))
+  q2 <- eval_code(q1, "b <- nrow(iris1)")
+  testthat::expect_identical(get_var(q2, "b"), 150L)
+})
+
+testthat::test_that("eval_code doesn't have access to environment where it's called", {
+  a <- 1L
+  q1 <- new_quosure("a <- 1", env = environment())
+  b <- 2L
+  testthat::expect_error(eval_code(q1, "d <- b"), "object 'b' not found")
+})
+
+testthat::test_that("@env in quosure is always a sibling of .GlobalEnv", {
+  q1 <- new_quosure()
+  testthat::expect_identical(parent.env(q1@env), parent.env(.GlobalEnv))
+
+  q2 <- eval_code(q1, "a <- 1L")
+  testthat::expect_identical(parent.env(q2@env), parent.env(.GlobalEnv))
+  q3 <- eval_code(q2, "b <- 2L")
+  testthat::expect_identical(parent.env(q3@env), parent.env(.GlobalEnv))
+})
+
+testthat::test_that("library have to be called separately before using function from package", {
+  q1 <- eval_code(new_quosure(), "library(checkmate)")
```

Comment:
Is the state before and after running this test definitely the same?

Reply:
content of the @env is the same but location of the @env changes.

Normally this is the environment hierarchy
```
- empty env
- base_env
- package1 env
- packageN env
- .GlobalEnv
```

When we create a quosure env is a parent of .GlobalEnv

```
- empty env
- base_env
- packageA env
- .... other packages env
- packageX env
- @env
- .GlobalEnv
```

When we call `library(Y)` within the eval_code, then Y becomes a parent of .GlobalEnv
```
- empty env
- base_env
- package A env
- .... other packages env
- package X env
- @env
- package Y env
- .GlobalEnv
```